skaffold/src/main/kotlin/com/google/container/tools/skaffold/SkaffoldExecutorService.kt

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright 2018 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.container.tools.skaffold
+
+import com.google.common.annotations.VisibleForTesting
+import com.intellij.openapi.components.ServiceManager
+import java.io.File
+import java.nio.file.Path
+import java.nio.file.Paths
+
+/**
+ * Interface for Skaffold execution service. This service builds and launches Skaffold process from
+ * the given set of standard flags and settings. [DefaultSkaffoldExecutorService] assumes Skaffold
+ * to be already installed and available.
+ */
+interface SkaffoldExecutorService {
+    companion object {
+        val instance
+            get() = ServiceManager.getService(SkaffoldExecutorService::class.java)!!
+    }
+
+    /**
+     * Creates Skaffold command line from the given settings and returns resulting launched
+     * Skaffold system process.
+     *
+     * @param settings Settings with flags supported by Skaffold.
+     * @return [SkaffoldProcess] with resulting process and the command line used to launch it.
+     */
+    fun executeSkaffold(settings: SkaffoldExecutorSettings): SkaffoldProcess
+
+    /**
+     * Execution mode for Skaffold, single run, continuous development, etc.
+     */
+    enum class SkaffoldExecutionMode(val modeFlag: String) {
+        SINGLE_RUN("run"),
+        DEV("dev")
+    }
+
+    /**
+     * Set of settings to control Skaffold execution, including flags and execution mode.
+     */
+    data class SkaffoldExecutorSettings(
```

Comment:
This documentation style makes it hard to see the actual code. Can you pull it up into @param values or have an explicit `constructor(...)?

Reply:
Sounds good, thanks, changed.